resources/lib/controller.py

```diff
@@ -743,13 +743,18 @@ def start_playback(args, api: API):
     # vo_drm_adaptive_dash
     # vo_drm_adaptive_hls
 
-    try:
-        url = req["streams"]["adaptive_hls"]
-        if args.subtitle in url:
-            url = url[args.subtitle]["url"]
-        elif args.subtitle_fallback in url:
-            url = url[args.subtitle_fallback]["url"]
+    try:        
+        if args.addon.getSetting("soft_subtitles") == "false":
+            url = req["streams"]["adaptive_hls"]
+            if args.subtitle in url:
+                url = url[args.subtitle]["url"]
+            elif args.subtitle_fallback in url:
+                url = url[args.subtitle_fallback]["url"]
+            else:
+                url = url[""]["url"]
         else:
+            #software subtitles
+            url = req["streams"]["multitrack_adaptive_hls_v2"]
```

Comment:
I'm curious, why did you pick this stream? :)

Reply:
This stream includes the embedded softsubs.

Reply:
An issue I see with this is that it only provides English subtitles, at least for me. It starts with no subtitles enabled, which is okay, but if I enable them I only have English ones and cannot switch to a different language. I have yet to find out how to make something like that work in Kodi, but if you already know...

Reply:
Huh, I'll see if there's a ways to request other languages in the stream.

Reply:
I've seen that method too, but it is for one language only. Which might be sufficient for a start.

What I personally would like to achieve and what I cannot figure out is how to tell the player about all the available subtitles or even video streams. In the OSD Menu for things like Netflix I can chose from all available audio and subtitle languages, but I can't figure out how to do that yet. :|

Reply:
I'm now setting the subtitle url that in this PR.

Reply:
I'm basically okay with this PR. I'd merge it and change a few things afterwards to handle the subtitle language fallback and cases where there are no appropriate subtitles.
If anyone finds out how to add multiple subtitles to the kodi player, please let me know :)

Reply:
@smirgol It's possible like this, but it has issues because the added subtitles don't have proper language names and instead show as GUIDs.

`item.setSubtitles([req["subtitles"][args.subtitle]["url"],req["subtitles"][args.subtitle_fallback]["url"]])`

Reply:
I thought I did try this yesterday to add all the subtitles, but somehow it didn't work. Need to look at it again.

And yes, it's not nice with the GUIDs. I have yet to find out how to make that nicer. 
My problem is, the documentation of Kodi is very lacking, plus you cannot define for which Kodi version you want the documentation. While it lists all possible methods, it doesn't explain at all how things are supposed to work together. It would totally have made sense to me to have 2 arguments with `setSubtitles` - the first for the label, the second for the URL, but Kodi says no and let's you guess how you can actually set the label. :|

Reply:
While it would solve the problem, I would find it very inconvenient to have such a popup before the start of each video.

It's a really annoying thing, the whole API does not make any sense to me and there is no good example that explains how they intended it to work. *sighs*

Reply:
So, all I could find out is that Kodi seems to rely solely on the filename of the subtitles to determine the language. This is why it shows a cryptic GUID, as this is the filename on Crunchyroll.

One way to solve it would be to download the subtitles, store them locally and rename the file.
That requires logic for cleaning up the downloaded subtitles, so they don't pile up and fill up your disk.
I really dislike this idea.

I have another very rough idea, where I need to play with that to see if that might work:
If I could somehow let the addon act as a proxy between Kodi and the original source of the subtitles, I could provide `setSubtitles` urls with filenames of my choice and when Kodi requests them, it points to the addon, which resolves the custom url, loads the matching data from Crunchyroll and passes them back to Kodi as the requested subtitle.
Something like 
`item->setSubtitles([addon://plugin.video.crunchroll/subtitle.en_EN.ass?source=REALURLHERE])`
It's just a wild idea, no clue if that might work.

Now, if Kodi simply would allow to specify the language for a subtitle stream and fallback to the filename guessing if none is provided, which it doesn't, life would be so much easier...

Reply:
Looks like there is a Kodi PR that would solve the issue. https://github.com/xbmc/xbmc/pull/17218

Reply:
Looks very promising and exactly what we would need, but unfortunately this won't be available in Leia/Matrix/Nexus, so for now we need a workaround and after fiddling with it for a bit, I think downloading the subs and renaming them is the way to go for. I'm a bit unsure if I should add that before trying to make a PR for MrKrabat's Repo. On the other hand I've already merged this PR and the solution is not optimal yet, due to the labelling of the Subtitle Language.